exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_recordable.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <map>
+#include <unordered_map>
+#include "nlohmann/json.hpp"
+#include "opentelemetry/sdk/common/attribute_utils.h"  // same as traces/attribute_utils
+#include "opentelemetry/sdk/logs/recordable.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+
+/**
+ * An Elasticsearch Recordable implemenation that stores the 10 fields of the Log Data Model inside
+ * a JSON object,
+ */
+class ElasticSearchRecordable final : public sdk::logs::Recordable
+{
+private:
+  // Define a JSON object that will be populated with the log data
+  nlohmann::json json_;
+
+  /**
+   * Converts a common::AttributeValue into a string, which is used for converting a
+   * common::AttributeValue into a string
+   */
+  const std::string ValueToString(const common::AttributeValue &value)
+  {
+    switch (value.index())
+    {
+      case common::AttributeType::TYPE_BOOL:
+        return (opentelemetry::nostd::get<bool>(value) ? "true" : "false");
+        break;
+      case common::AttributeType::TYPE_INT:
+        return std::to_string(opentelemetry::nostd::get<int>(value));
+        break;
+      case common::AttributeType::TYPE_INT64:
+        return std::to_string(opentelemetry::nostd::get<int64_t>(value));
+        break;
+      case common::AttributeType::TYPE_UINT:
+        return std::to_string(opentelemetry::nostd::get<unsigned int>(value));
+        break;
+      case common::AttributeType::TYPE_UINT64:
+        return std::to_string(opentelemetry::nostd::get<uint64_t>(value));
+        break;
+      case common::AttributeType::TYPE_DOUBLE:
+        return std::to_string(opentelemetry::nostd::get<double>(value));
+        break;
+      case common::AttributeType::TYPE_STRING:
+      case common::AttributeType::TYPE_CSTRING:
+        return opentelemetry::nostd::get<opentelemetry::nostd::string_view>(value).data();
+        break;
+      default:
+        return "Invalid type";
+        break;
+    }
+  }
+
+public:
+  /**
+   * Set the severity for this log.
+   * @param severity the severity of the event
+   */
+  void SetSeverity(opentelemetry::logs::Severity severity) noexcept override
+  {
+    // Convert the severity enum to a string
+    static opentelemetry::nostd::string_view severityStringMap_[25] = {
```

Comment:
nit: this map is already included in https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/logs/severity.h#L65

Reply:
cool I didn't realize the recordable PR had this, definitely makes it simpler :)